api/nodes/serializers.py

```diff
@@ -46,7 +63,14 @@ def get_absolute_url(self, obj):
         return obj.absolute_url
 
     def get_node_count(self, obj):
-        return len(obj.nodes)
+        request = self.context['request']
+        user = request.user
+        if user.is_anonymous():
+            auth = Auth(None)
+        else:
+            auth = Auth(user)
+        nodes = [node for node in obj.nodes if node.can_view(auth) and node.primary]
```

Comment:
@samanehsan, do you have any thoughts on the feasibility of Sam's suggestion, since you implemented it? I expect probably it won't work, but I don't know if you tried something like it early on. I figure it's worth checking before I poke at it too much.